icu4j/main/core/src/main/java/com/ibm/icu/text/RuleBasedBreakIterator.java

```diff
@@ -99,17 +100,62 @@ public static RuleBasedBreakIterator getInstanceFromCompiledRules(InputStream is
      *
      * @param bytes a buffer supplying the compiled binary rules.
      * @param phraseBreaking a flag indicating if phrase breaking is required.
+     * @param dxValues Dictionary break script exclusions.
      * @throws IOException if there is an error while reading the rules from the buffer.
      * @see    #compileRules(String, OutputStream)
      * @internal
      */
     /* package-potected */ static RuleBasedBreakIterator getInstanceFromCompiledRules(
-            ByteBuffer bytes, boolean phraseBreaking) throws IOException {
+            ByteBuffer bytes, boolean phraseBreaking, String dxValues) throws IOException {
         RuleBasedBreakIterator instance = getInstanceFromCompiledRules(bytes);
         instance.fPhraseBreaking = phraseBreaking;
+        instance.fDX = makeExcludedDictionaryBreakUnicodeSet(dxValues);
         return instance;
     }
 
+    /**
+     * Crate a UnicodeSet for the Dictionary Break Script Exclusions.
+     * @param dxValues Dictionary break script exclusions, a string of Script code joined by "-".
+     * @internal
+     */
+    private static UnicodeSet makeExcludedDictionaryBreakUnicodeSet(
+        String dxs) {
+        if (dxs == null) {
+            return null;
+        }
+        if (dxs.length() % 5 != 4) {
+            throw new IllegalArgumentException("Incorrect value for dx key: " + dxs);
+        }
+        // Change from "thai" to "[[:scx=thai:]]" or "thai-arab" to "[[:scx=thai:][:scx=arab:]]"
+        StringBuilder builder = new StringBuilder("[");
+        int items = 1 + (dxs.length() / 5);
+        for (int i = 0; i < items; i++) {
+            if (i > 0 && dxs.charAt(i*5-1) != '-') {
+                throw new IllegalArgumentException("Incorrect value for dx key: " + dxs);
+            }
+            String script = dxs.substring(i*5, i*5+4);
+            // Special handling of zyyy
```

Comment:
Needs a test case also.

Reply:
but we need to take care the case of "en-u-dx-thai-zyyy" or "en-u-dx-thai-hani-zyyy", etc too right?

Reply:
My point is " if the entire dxValues value equals (case insensitive) "-zyyy"" is not good enough because we may have 
 "en-u-dx-thai-zyyy" or "en-u-dx-thai-hani-zyyy" which the type is "thai-hani-zyyy" not just "zyyy"

Reply:
oh. Just saw what you landed in https://github.com/unicode-org/cldr/pull/3411/files that make sense. Sorry. Ignore my previous comments.